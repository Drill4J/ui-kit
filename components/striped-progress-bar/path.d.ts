export declare const path = "M13.629 0l-12 12H0V7.973L7.973 0h5.656zm7.68 0L9.308 12H3.652l12-12h5.657zm7.68 0L16.988 12H11.33l12-12h5.658zm7.68 0L24.668 12h-5.656l12-12h5.657zm7.68 0L32.348 12h-5.656l12-12h5.657zm7.68 0L40.028 12h-5.656l12-12h5.657zm7.68 0L47.708 12H42.05l12-12h5.658zm7.68 0L55.388 12H49.73l12-12h5.658zm7.68 0L63.068 12h-5.656l12-12h5.657zm7.68 0L70.748 12h-5.656l12-12h5.657zm7.68 0L78.428 12h-5.656l12-12h5.657zm7.68 0L86.108 12H80.45l12-12h5.658zm7.68 0L93.788 12H88.13l12-12h5.658zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.24 0l-12.001 12h-5.656l12-12h5.657zm7.8 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm8.08 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.656l12-12h5.657zM841 0v2.149l-9.852 9.85h-5.657l12-12L841 0zm0 4.172v5.656L838.828 12h-5.657L841 4.172zM5.829 0l-5.83 5.829V.173L.174 0h5.656z";
