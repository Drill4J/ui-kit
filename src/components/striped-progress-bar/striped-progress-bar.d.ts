interface Props {
  className?: string;
  value?: string;
  type: 'primary' | 'secondary';
}
export declare const StripedProgressBar: {
  ({ className, value, type }: Props): JSX.Element;
  displayName: string;
  __docgenInfo: {
    description: string;
    displayName: string;
    props: {
      className: {
        defaultValue: null;
        description: string;
        name: string;
        required: boolean;
        type: {
          name: string;
        };
      };
      value: {
        defaultValue: {
          value: string;
        };
        description: string;
        name: string;
        required: boolean;
        type: {
          name: string;
        };
      };
      type: {
        defaultValue: null;
        description: string;
        name: string;
        required: boolean;
        type: {
          name: string;
        };
      };
    };
  };
};
export {};
