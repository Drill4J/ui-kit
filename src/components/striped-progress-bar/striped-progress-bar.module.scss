@import '../../style-guide/index';

.striped-progress-bar {
  display: block;
  border-radius: 0 4px 4px 0;
  transition: width 2000ms ease-in;

  &--type {
    &--overlapping {
      fill: map-get($data-visualization, overlapping);
      transform: rotate(180deg);
      border-radius: 4px 0 0 4px;
    }

    &--scope {
      fill: map-get($data-visualization, scope-cover);
    }
  }
}
