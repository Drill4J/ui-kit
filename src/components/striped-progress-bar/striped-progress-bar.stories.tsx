import React from 'react';
import { storiesOf } from '@storybook/react';

import { StripedProgressBar } from './striped-progress-bar';

storiesOf('StripedProgressBar', module)
  .add('StripedProgressBar with change coverage panel', () => {
    const [coverage, setCoverage] = React.useState(0);
    const handleChange = ({ target: { value } }: React.ChangeEvent<HTMLInputElement>) => {
      setCoverage(Number(value));
    };
    return (
      <div style={{ padding: '40px' }}>
        <div style={{
          display: 'grid', placeItems: 'center', width: '500px', height: ' 500px',
        }}
        >
          <StripedProgressBar value={`${coverage}%`} type="primary" />
        </div>
        <input type="range" min="0" max="100" value={coverage} onChange={handleChange} />
      </div>
    );
  });
