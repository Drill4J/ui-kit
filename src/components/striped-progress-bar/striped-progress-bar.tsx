import { BEM } from '@redneckz/react-bem-helper';

import { path } from './path';

import styles from './striped-progress-bar.module.scss';

interface Props {
  className?: string;
  value?: string;
  type: 'primary' | 'secondary';
}

const stripedProgressBar = BEM(styles);

export const StripedProgressBar = stripedProgressBar(({ className, value, type }: Props) => (
  <svg className={className} type={type} width={value} height="12" data-test={`striped-progress-bar:${type}`}>
    <path
      d={path}
      fillRule="nonzero"
      stroke="none"
      strokeWidth="1"
      strokeDashoffset={value}
    >
      <animateTransform
        attributeName="transform"
        type="translate"
        from="-100"
        to="1"
        additive="replace"
        dur="5"
        repeatCount="indefinite"
      />
    </path>
  </svg>
));
