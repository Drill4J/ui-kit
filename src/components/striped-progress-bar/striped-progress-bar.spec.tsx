import React from 'react';
import renderer from 'react-test-renderer';

import { StripedProgressBar } from './striped-progress-bar';

describe('StripedProgressBar', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<StripedProgressBar value="50px" type="primary" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
