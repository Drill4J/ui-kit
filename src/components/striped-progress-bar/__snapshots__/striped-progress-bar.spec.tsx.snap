// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StripedProgressBar should match snapshot 1`] = `
<svg
  className="striped-progress-bar striped-progress-bar--type--overlapping"
  data-test="striped-progress-bar:OVERLAPPING"
  height="12"
  type="OVERLAPPING"
  width="50px"
>
  <path
    d="M13.629 0l-12 12H0V7.973L7.973 0h5.656zm7.68 0L9.308 12H3.652l12-12h5.657zm7.68 0L16.988 12H11.33l12-12h5.658zm7.68 0L24.668 12h-5.656l12-12h5.657zm7.68 0L32.348 12h-5.656l12-12h5.657zm7.68 0L40.028 12h-5.656l12-12h5.657zm7.68 0L47.708 12H42.05l12-12h5.658zm7.68 0L55.388 12H49.73l12-12h5.658zm7.68 0L63.068 12h-5.656l12-12h5.657zm7.68 0L70.748 12h-5.656l12-12h5.657zm7.68 0L78.428 12h-5.656l12-12h5.657zm7.68 0L86.108 12H80.45l12-12h5.658zm7.68 0L93.788 12H88.13l12-12h5.658zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.24 0l-12.001 12h-5.656l12-12h5.657zm7.8 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm8.08 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.656l12-12h5.657zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.657l12-12h5.658zm7.68 0l-12.001 12h-5.656l12-12h5.657zM841 0v2.149l-9.852 9.85h-5.657l12-12L841 0zm0 4.172v5.656L838.828 12h-5.657L841 4.172zM5.829 0l-5.83 5.829V.173L.174 0h5.656z"
    fillRule="nonzero"
    stroke="none"
    strokeDashoffset="50px"
    strokeWidth="1"
  >
    <animateTransform
      additive="replace"
      attributeName="transform"
      dur="5"
      from="-100"
      repeatCount="indefinite"
      to="1"
      type="translate"
    />
  </path>
</svg>
`;
